src/multihash.rs

```diff
@@ -173,6 +173,45 @@ impl<const S: usize> Multihash<S> {
             .expect("writing to a vec should never fail");
         bytes
     }
+
+    /// Truncates the multihash to the given size. It's up to the caller to ensure that the new size
+    /// is safe to use.
+    ///
+    /// If the new size is larger than the current size, this method does nothing.
+    ///
+    /// ```
+    /// use multihash::{Code, MultihashDigest};
+    ///
+    /// let hash = Code::Sha3_256.digest(b"Hello world!").truncate(20);
+    /// ```
+    pub fn truncate(&self, size: u8) -> Self {
+        let mut mh = *self;
+        mh.size = mh.size.min(size);
+        mh
+    }
+
+    /// Resizes the backing multihash buffer. This function fails if the hash digest is larger than
+    /// the target size.
+    ///
+    /// ```
+    /// use multihash::{Code, MultihashDigest, MultihashGeneric};
+    ///
+    /// let hash = Code::Sha3_256.digest(b"Hello world!");
+    /// let large_hash: MultihashGeneric<32> = hash.resize().unwrap();
+    /// ```
+    pub fn resize<const R: usize>(&self) -> Result<Multihash<R>, Error> {
```

Comment:
This should ideally be a compiler error instead of a runtime error as we should know both the digest size and the resize size(?) at compile time. the heapless crate has a [neat hack](https://github.com/japaric/heapless/blob/master/src/sealed.rs) to make the compiler notice this in stable rust

Reply:
A playground link demonstrating https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=47e71bb4e1fbe3dbd24ac182918c1d93

Reply:
We know the buffer size, but not the _digest_ size. It should be possible to resize from a `Multihash<64>` to a `Multihash<32>` if the digest fits.

Reply:
we should know the digest size! even with blake3 we limited it to 32 bytes to ensure we know how big it is.